### ISSUE TITLE
Modal: improve accessibility markup

### ISSUE THREAD

**Issue by simonihmig** [kaliber5/ember-bootstrap#674]
See http://getbootstrap.com/docs/4.1/components/modal/#accessibility

**Comment by collinsnji** [comment]
Can you please provide details on what you need improved? That'll be much appreciated

**Comment by simonihmig** [comment]
Sure:
* `aria-labelledby` is currently unused, and should be set like specified in the Bootstrap docs
* `role="document"` is missing for the `.modal-dialog`
* add tests for the above points

**Comment by valentinaperic** [comment]
I can work on this! 👍

**Comment by simonihmig** [comment]
Closed by #688
